### ISSUE TITLE
Auto-Type: more info on entries in entry selection dialog

### ISSUE THREAD

**Issue by anetri** [keepassxreboot/keepassxc#1416]
It would be great, if you could add more info on entries displayed in entry selection dialog for Auto-Type.

## Expected Behavior
More fields could be displayed in "Select entry to Auto-Type" list for each entry, e.g. URL or Notes, or other - possibly configurable.

## Current Behavior
Only Group, Title and Username are displayed.

## Possible Solution
Add more columns to display more fields for entries. Or at least display more info in tooltip when hoovering over an entry. 

## Context
I am using KeePassXC to store credentials for many similar (or same) webpages running on different servers (e.g. dev, test, prod environments). I prefer to keep all entries in the same group, use same username but different passwords and therefore using autotype shows me several equal entries with no possibility to find the right on (except for maybe order or try/fail approach). This is annoying and this feature would be very useful and hopefully easy to implement. Thanks.

## Debug Info
KeePassXC - Version 2.2.4
Revision: 4723f66

Libraries:
- Qt 5.9.2
- libgcrypt 1.8.1

Operating system: Windows 7 SP 1 (6.1)
CPU architecture: x86_64
Kernel: winnt 6.1.7601

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey

**Comment by TheZ3ro** [comment]
Are you using different entry (dev, production, etc) with the same title and from the same group?

You should consider adding a `-dev` label in the title of dev entry or using a different group for them.

Anyway, adding this to the wishlist

**Comment by anetri** [comment]
Yes, all entries have same Title, same Username and are in the same group.
Adding anything into title would cause the entry not matching to window title, which is not good.
Using different groups is a possible solution, but is impractical and hard to maintain when too many groups are created.
Anyway, wishlist is fine. :wink:

**Comment by DJCrashdummy** [comment]
i experienced a pretty similar problem while having more than one database open... when having similar entries in the same groups, it is not so easy to differentiate between these entries! :confused: 
so at least an (optional) column with the database-name would solve this problem.

**Comment by pwilliamson19** [comment]
I have the same problem. This happens with a few entries and makes it impossible to select the right one. So this version is useless for me. Back to Keepasspro, or something else.....
Even displaying the first line of the Notes field would help - I put a comment in there to identify the site. And displaying the sequence seems to serve no useful purpose (except possibly debugging?), and takes up a lot of space. Ideally the content of the window should be configureable.

**Comment by l0f4r0** [comment]
I support this feature request because I'm impacted as well (for some websites, I cannot decide which entry to choose because there are not enough discriminating info/columns displayed currently)...

**Comment by droidmonkey** [comment]
Just add more information to the entry's title in the meantime.

**Comment by l0f4r0** [comment]
> Just add more information to the entry's title in the meantime.

I understand but as said by @anetri this would break the auto-type feature because the entry title would not be recognized anymore...

**Comment by droidmonkey** [comment]
The entry title has nothing to do with matching auto type.

**Comment by l0f4r0** [comment]
> The entry title has nothing to do with matching auto type.

What? Can you elaborate please?? If entry title has nothing to do with auto-type matching, what is taken into account for that purpose?

**Comment by droidmonkey** [comment]
Everything in the red box:

![image](https://user-images.githubusercontent.com/2809491/55516658-e1087d80-563b-11e9-9e9f-4af1adc86952.png)

**Comment by anetri** [comment]
> The entry title has nothing to do with matching auto type.

Wow, when I opened this issue, I thought it is only entry title that is used. I somehow missed the "Window Associations" of entry, therefore all of my entries had this section empty and only entry title was used when auto-type matching (also at that time, there was no setting "Use entry title ...").

So I guess my feature request is therefore even less important or even unnecessary at all. Thanks for advise... :wink:

**Comment by l0f4r0** [comment]
Great @droidmonkey, I didn't pay so much attention to that auto-type tab setting. It does the job indeed! :)
It forces you to edit your title to have something more meaningful and then edit your auto-type triggering feature for your entry (compared to only displaying some more columns in auto-type popup when triggered) but it's OK in the meantime ;)
Thanks!

**Comment by droidmonkey** [comment]
Great, I think we should still show more information in the Auto-Type selection window, or at least have the ability to show more.

**Comment by pwilliamson19** [comment]
I have followed this with some interest, and thanks to this info I have now got KeepassXC behaving in a useable fashion. Thanks particularly to droidmionkey! However I woulds still like to see the selection window more configurable. Group and Sequence are a waste of space in this context, as far as I am concerned. A configurable window as per KeepassPro would do the job.
Also - KeepassPro has lots of help for setting your own autotype sequence. I can't find anything comparable in KeepassXC which makes it very awkward to set up your own sequence if it is anything other than very simple (and I have one or two). A similar feature in KeepassXC would be most helpful.
